src/navigationStore.js

```diff
@@ -190,19 +193,40 @@ function createNavigationOptions(params) {
     return res;
   };
 }
-
-function createWrapper(Component, wrapBy) {
+function originalRouteName(routeName) {
+  if (routeName.startsWith('_')) {
+    return routeName.substring(1);
+  }
+  return routeName;
+}
+// eslint no-param-reassign: "error"
+function createWrapper(Component, wrapBy, store: NavigationStore) {
   if (!Component) {
     return null;
   }
   const wrapper = wrapBy || (props => props);
-  function wrapped({ navigation, ...props }) {
-    return <Component {...props} navigation={navigation} {...navigation.state.params} name={navigation.state.routeName} />;
+
+  class Wrapped extends React.Component {
+    componentDidMount() {
+      const navigation = this.props.navigation;
+      if (this.ref) {
+        store.addRef(originalRouteName(navigation.state.routeName), this.ref);
+      }
+    }
+    componentWillUnmount() {
+      const navigation = this.props.navigation;
+      this.ref = null;
+      store.deleteRef(originalRouteName(navigation.state.routeName));
+    }
+    render() {
+      const navigation = this.props.navigation;
+      return <Component ref={ref => (this.ref = ref)} {...this.props} {...navigation.state.params} name={navigation.state.routeName} />;
```

Comment:
Good point. However making Screens as pure function is not good idea - hot reload will not work for them...